src/Http/Requests/Baskets/CreateLinesRequest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace GetCandy\Api\Http\Requests\Baskets;
+
+use GetCandy\Api\Http\Requests\FormRequest;
+
+class CreateLinesRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        $rules = [
+            'basket_id' => 'sometimes|hashid_is_valid:baskets',
+            'variants' => 'array|unique_lines',
+        ];
+
+        $variants = app('api')->productVariants()->getByHashedIds(
```

Comment:
There is a method available `getDecodedIds` which doesn't touch the database and should give back what you need here.

Reply:
Okay great!

Reply:
I think the database accessing is required, as further on, it uses data from `$variants`, such as min_qty and min_batch.

If you think think all of that is unnecessary, let me know and we can work out how to refactor it. I pulled it from the existing `CreateRequest.php`